examples/js/archive.js

```diff
@@ -178,16 +189,100 @@ function getImageName(imageURL) {
   return imageName;
 }
 
+function extractJsonFromUrlParams(url) { 
+  const startIndex = url.lastIndexOf('=');
+  const jsonDownloadURL = url.slice(startIndex + 1);
+
+  return jsonDownloadURL;
+}
+
+function isJsonDetected(url) {
+  if (url.includes('?json=')) {
+    const startIndex = url.lastIndexOf('.');
+    const fileExtension = url.slice(startIndex + 1);
+
+    if (fileExtension === 'json') {
+      console.log('JSON found in map shareable link'); // left here purposely
+      return true;
+    } 
+  }
+
+  return false;
+}
+
+function createMap (imageURL, options, newImage = false) {
+  let image;
+  
+  if (newImage) { 
+    image = L.distortableImageOverlay(
+      imageURL,
+      {tooltipText: options.tooltipText}
+    );
+  } else {
+    image = L.distortableImageOverlay(
+      imageURL,
+      {
+        height: options.height,
+        tooltipText: options.tooltipText,
+        // corners: options.corners, <== uncomment this to see the effect of the corners
+      }
+    );
+  }
+
+  map.imgGroup.addLayer(image);
+};
+
+// Reconstruct Map from JSON
+document.addEventListener('DOMContentLoaded', async (event) => {
+  if (mapReconstructionMode) {
+    const url = location.href;
+    
+    if (isJsonDetected(url)) {
+      const jsonDownloadURL = extractJsonFromUrlParams(url); 
+
+      if (jsonDownloadURL) {
+        const imageCollectionObj = await map.imgGroup.recreateImagesFromJsonUrl(jsonDownloadURL); 
+        const avg_cm_per_pixel = imageCollectionObj.avg_cm_per_pixel; // this is made available here for future use
+      
+        if (imageCollectionObj.imgCollectionProps.length > 1) {
+          let imageURL;
+          let options;
+      
+          imageCollectionObj.imgCollectionProps.forEach((imageObj) => {
+            imageURL = imageObj.src;
+            options = {
+              height: imageObj.height,
+              tooltipText: imageObj.tooltipText,
+              corners: imageObj.nodes,
+            };
+            createMap(imageURL, options, false);
+          });
+      
+          return;
+        }
+
+        const imageObj = imageCollectionObj.imgCollectionProps[0];
```

Comment:
I'm sorry, I'm not quite following this section, why do we need to have two sections where we create an overlay? Maybe it needs an explanatory comment? Thanks!

Also, could we move all of this into recreateImagesFromJsonUrl?

Reply:
**Two sections for overlay:** first section is needed because of the need to iteratively create overlays in cases where multiple images are to be retrieved from json file. The second section comes in handy when only one overlay is to be retrieved from the json file.

**Move to recreateImagesFromJsonUrl:** Like you suggested earlier which I have since agreed to and refactored the code accordingly (i.e., to it current state), I think this is unnecessary as it's a user application responsibility. The library simply supplies the data required to reconstruct overlays to their previous state, the user then has the flexibility to place the overlays however he wants using available API.  What do you think?

Reply:
Ok that reasoning sounds good to me! Let me know and I can hit merge as soon as you like. Thanks!!

Reply:
Hi @jywarren, you may now consider for merging. Many thanks!